DynamicTablesPkg/Include/ArmNameSpaceObjects.h

```diff
@@ -420,6 +420,48 @@ typedef struct CmArmGenericWatchdogInfo {
   UINT32    Flags;
 } CM_ARM_GENERIC_WATCHDOG_INFO;
 
+/** The ARM_PCI_OSC_INFO union/struct describes the features that FW will
+ * grant native control over via _OSC method.
+*/
+typedef union PciOscControlBufferInfo {
+  struct {
+    /// Allow OS Native control of Hot Plug if set
+    UINT32     AllowNativeHotPlugControl : 1;
+
+    /// Allow Native SHPC Control
+    UINT32     AllowNativeShpcControl : 1;
+
+    /// Allow Native Power Management Events Control
+    UINT32     AllowNativePmeControl : 1;
+
+    /// Allow Native Advanced Error Reporting Control
+    UINT32     AllowNativeAerControl : 1;
+
+    /// Allow Native Capability Structure Control
+    UINT32     AllowNativeCapabilityControl : 1;
+
+    /// Allow Native Latency Tolerance Reporting Control
+    UINT32     AllowNativeLtrControl : 1;
+
+    /// Firmware sets this bit to 1 to indicate that it 
+    /// will suppress error notification caused by surprise hot remove
+    /// (Completion Time Out errors)
+    UINT32     SupressSurpriseHotRemoveErrors : 1;
+
+    /// Allow Native Downstream Port Containment Control
+    UINT32     AllowNativeDpcControl : 1;
+
+    /// Allow Native Completion Timeout Control
+    UINT32     AllowNativeCompletionTimeoutControl : 1;
+
+    /// Allow Native System Firmware Intermediary Configuration Control
+    UINT32     AllowNativeSfiControl : 1;
+
+    UINT32     Reserved22 : 22;
+  };
+  UINT32     AsUint32;

```

Comment:
Hi,
I never had this issue, but I thought it was not advised to use bit-fields like that as it is not guaranteed that each field will conserve its position (i.e. AllowNativeHotPlugControl might not be mapped to the first bit of the resulting uint32).
I usually see bit-fields used as masks:
`#define PIC_OSC_ALLOW_NATIVE_HOT_PLUG_CONTROL      0x01`

Reply:
This is fair, I think in practice this will work, but the C programming language book does say: "Almost everything about fields is implementation-dependent." It is likely safer to define the bitfields as Pierre showed.

Reply:
Thanks for the note, updated to use #define masks instead of a struct/union